samples/TerraFX/Graphics/HelloModelViewProjection.cs

```diff
@@ -0,0 +1,278 @@
+
+            currentGraphicsContext.BeginFrame();
+            ulong vertexBufferSize = 0;
+            ulong indexBufferSize = 0;
+
+            for (var i = 0; i < _meshes.Count; ++i)
+            {
+                var mesh = _meshes[i];
+                vertexBufferSize += (ulong)(SizeOf<PosNormColorVertex>() * mesh.Vs.Length);
+                indexBufferSize += (ulong)(SizeOf<uint>() * 3 * mesh.Ts.Length);
+            }
+
+            using var vertexStagingBuffer = graphicsDevice.MemoryAllocator.CreateBuffer(GraphicsBufferKind.Default, GraphicsResourceCpuAccess.CpuToGpu, vertexBufferSize);
+            using var indexStagingBuffer = graphicsDevice.MemoryAllocator.CreateBuffer(GraphicsBufferKind.Default, GraphicsResourceCpuAccess.CpuToGpu, indexBufferSize);
+            _constantBuffer = graphicsDevice.MemoryAllocator.CreateBuffer(GraphicsBufferKind.Constant, GraphicsResourceCpuAccess.CpuToGpu, 64 * 1024);
+            _vertexBuffer = graphicsDevice.MemoryAllocator.CreateBuffer(GraphicsBufferKind.Vertex, GraphicsResourceCpuAccess.GpuOnly, vertexBufferSize);
+            _indexBuffer = graphicsDevice.MemoryAllocator.CreateBuffer(GraphicsBufferKind.Index, GraphicsResourceCpuAccess.GpuOnly, indexBufferSize);
+
+            for (var i = 0; i < _meshes.Count; ++i)
+            {
+                _meshPrimitives.Add(CreateMeshPrimitive(i, currentGraphicsContext, vertexStagingBuffer, indexStagingBuffer));
+            }
+
+            currentGraphicsContext.EndFrame();
+
+            graphicsDevice.Signal(currentGraphicsContext.Fence);
+            graphicsDevice.WaitForIdle();
+        }
+
+        public override void Cleanup()
+        {
+            for (var i = 0; i < _meshPrimitives.Count; i++)
+            {
+                _meshPrimitives[i]?.Dispose();
+            }
+
+            _constantBuffer?.Dispose();
+            _vertexBuffer?.Dispose();
+            _indexBuffer?.Dispose();
+            base.Cleanup();
+        }
+
+        // ----- methods.protected -----
+
+        protected override void Draw(GraphicsContext graphicsContext)
+        {
+            foreach (var meshPrimitive in _meshPrimitives)
+            {
+                graphicsContext.Draw(meshPrimitive);
+            }
+
+            base.Draw(graphicsContext);
+        }
+
+        protected override unsafe void Update(TimeSpan delta)
+        {
+            const float RotationSpeed = 1f;
+
+            var angle = _meshRotationAngle;
+            var angleDelta = (float)(RotationSpeed * delta.TotalSeconds);
+            angle += angleDelta;
+            angle %= MathF.Tau;
+
+            _meshRotationAngle = angle;
+            var sin = MathF.Sin(angle);
+            var cos = MathF.Cos(angle);
+
+            var camToWorld = Transform.AddRotationAroundSourceY(Transform.Identity, angle)
+                .WithRotCenterTarget(new Vector3(cos, 0, sin));
+            _camera = _camera.WithToWorld(camToWorld);
+
+            if (_meshPrimitives.Count > 0) { // box
```

Comment:
Allman style braces are the default, except for a few cases like lambda expressions.
```suggestion
            if (_meshPrimitives.Count > 0)
            {
```

Reply:
It's also not clear why this is needed. Are both meshes not guaranteed to be present and if so, what logic is controlling that?

Reply:
agreed on Allman style braces. Usually I do a reformat whole document in VisualStudio at the end which fixes that. Seems I forgot that here.  Fixed now.

Reply:
`if (count)`: sorry that was leftover temp code from before having all vertex indices correct. No longer needed.  I removed it.